android/capacitor/src/main/java/com/getcapacitor/Bridge.java

```diff
@@ -786,10 +787,10 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
                 if (validatePermissions(plugin.getInstance(), savedPermissionCall, permissions, grantResults)) {
                     // handle request permissions call
                     if (savedPermissionCall.getMethodName().equals("requestPermissions")) {
-                        savedPermissionCall.resolve(plugin.getInstance().getPermissionStates());
+                        plugin.getInstance().checkPermissions(savedPermissionCall);
                     } else {
                         // handle permission requests by other methods on the plugin
-                        plugin.getInstance().onRequestPermissionsResult(savedPermissionCall, requestCode, permissions, grantResults);
+                        plugin.getInstance().onRequestPermissionsResult(savedPermissionCall, getPermissionStates(plugin.getInstance()));
```

Comment:
Do we want to call the plugin instance version?

```suggestion
                        plugin.getInstance().onRequestPermissionsResult(savedPermissionCall, plugin.getInstance().getPermissionStates());
```